### PR TITLE
devSerialIO refactor

### PR DESCRIPTION
**Pull request by mha1**
This is work is based on @CapnBry's push to remove redundant code between devSerialIO and devSerial1IO. 

@CapnBry Thanks for the ideas and discussion.

Changes:
- have a single context based SerialIO driver handling both Serial0 and Serial1 devices
- removed redundant code
- removed devSerial1
- removed redundant variable initializations
- moved Serial1 config to unused padding byte (retaining config struct size)

### CODE REVIEW

File: src/lib/CONFIG/config.h [hunk 1]
```diff
@@ -201,7 +201,9 @@ typedef union {
 typedef struct __attribute__((packed)) {
     uint32_t    version;
     uint8_t     uid[UID_LEN];
-    uint8_t     unused_padding[2];
```
Comment (pacaj2am) [inline src/lib/CONFIG/config.h:204]: Shouldn't this change bump config version and add mapping as well _legacy structure?

Comment (pkendall64) [reply src/lib/CONFIG/config.h:204]: No, as these are unused they will be 0 by default which is NONE.
And this second serial selection is new for the master branch.

Comment (pacaj2am) [reply src/lib/CONFIG/config.h:204]: might have separate question - deep dived into the logic - why serial 0 is having default CRSF and Serial1 has NONE?

Comment (mha1) [reply src/lib/CONFIG/config.h:204]: The addition of a second serial interface allowing to run a second protocol in parallel is a true extension. Nothing was changed to the way serial0 was working. With serial1 initialized to NONE 3.5 exhibits no difference to the way 3.4 and previous versions present a virgin receiver to the user. CRSF is still the default protocol ready to talk to an FC just like the founding fathers intended it (I guess).

**Review by pkendall64 — APPROVED** [review]

**Review by CapnBry — APPROVED** [review]
Great! Nice to reuse the same code instead of duplicating the source files and having to maintain both going forward. Thanks for revisiting this and making me happy.

Tested on Betafpv SuperP CRSF + Nothing and CRSF + SBUS.

File: src/src/rx-serial/devSerialIO.cpp [hunk 2]
```diff
@@ -8,40 +8,86 @@
 #include "CRSF.h"
 #include "config.h"
 
+#define NO_SERIALIO 1000
```
[the remark this answers is not in the document]
Comment (mha1) [reply src/src/rx-serial/devSerialIO.cpp]: Not at all. Done, sealed and delivered.

Thanks for your review.

**Comment by CapnBry** [comment]
Awesome, thanks for putting up with my nitpicking

**Comment by mha1** [comment]
my pleasure

**Pull request merged**